satpy/composites/__init__.py

```diff
@@ -1041,12 +1042,22 @@ def __init__(self, name, filename=None, url=None, known_hash=None, area=None,
     def _get_cache_filename_and_url(filename, url):
         if filename is not None:
             filename = os.path.expanduser(os.path.expandvars(filename))
+
+            if not os.path.isabs(filename):
+                data_dir = satpy.config.get('data_dir')
+                filename = os.path.join(data_dir, filename)
+                if not os.path.exists(filename):
+                    raise ValueError('Can not find image {} for '
+                                     'StaticImageCompositor'.format(filename))
```

Comment:
I think this changes the overall behavior of how this is supposed to work. The `filename` is meant to allow the user to customize how the file is stored in the data_dir. If you add the data_dir here and it doesn't exist, it should still allow the `url` to be used in `register_data_files`. I think if the absolute path with data_dir doesn't exist then the original filename should be used and we should depend on the error message at the end of this method.

Thoughts?

Reply:
Updated the logic according to the discussion of use cases in slack.